src/backend/cdb/cdbgang.c

```diff
@@ -444,7 +444,7 @@ disconnectAndDestroyGang(Gang *gp)
 		elog(DEBUG2, "Warning: disconnectAndDestroyGang called on an %s gang",
 			 gp->active ? "active" : "allocated");
 
-	if (gp->gang_id < 1 || gp->gang_id > 100000000 || gp->type > 10 || gp->size > 100000)
+	if (gp->gang_id < 1 || gp->gang_id > 100000000 || gp->type > GANGTYPE_PRIMARY_WRITER || gp->size > 100000)
```

Comment:
check >10 is meaningless. We can also add `GANGTYPE_MAX` then check with `gp->type >= GANGTYPE_MAX`. Also, we can check with `gp-type < GANGTYPE_MAX`. All of these are equivalent.